packages/arb-token-bridge-ui/src/hooks/useCompleteMultiChainTransactions.ts

```diff
@@ -0,0 +1,331 @@
+  }
+
+  return getStandardizedTimestamp(tx.timestamp ?? '0')
+}
+
+function sortByTimestampDescending(
+  a: DepositOrWithdrawal,
+  b: DepositOrWithdrawal
+) {
+  return getStandardizedTimestampByTx(a) > getStandardizedTimestampByTx(b)
+    ? -1
+    : 1
+}
+
+const multiChainFetchList: { parentChain: ChainId; chain: ChainId }[] = [
+  {
+    parentChain: ChainId.Mainnet,
+    chain: ChainId.ArbitrumOne
+  },
+  // {
+  //   parentChain: ChainId.Mainnet,
+  //   chain: ChainId.ArbitrumNova
+  // },
+  // Testnet
+  {
+    parentChain: ChainId.Goerli,
+    chain: ChainId.ArbitrumGoerli
+  },
+  {
+    parentChain: ChainId.Sepolia,
+    chain: ChainId.ArbitrumSepolia
+  }
+  // Orbit
+  // {
+  //   parentChain: ChainId.ArbitrumGoerli,
+  //   chain: ChainId.XaiTestnet
+  // }
+  // {
+  //   parentChain: ChainId.ArbitrumSepolia,
+  //   chain: ChainId.StylusTestnet
+  // }
+]
+
+function isWithdrawalFromSubgraph(
+  tx: Withdrawal
+): tx is FetchWithdrawalsFromSubgraphResult & AdditionalProperties {
+  return tx.source === 'subgraph'
+}
+
+function isDeposit(tx: DepositOrWithdrawal): tx is Deposit {
+  return tx.direction === 'deposit'
+}
+
+async function transformTransaction(tx: DepositOrWithdrawal) {
+  const parentChainProvider = getProvider(tx.parentChainId)
+  const chainProvider = getProvider(tx.chainId)
+
+  if (isDeposit(tx)) {
+    return transformDeposit(
+      await updateAdditionalDepositData({
+        depositTx: tx,
+        l1Provider: parentChainProvider,
+        l2Provider: chainProvider
+      })
+    )
+  }
+
+  let withdrawal: L2ToL1EventResultPlus | undefined
+
+  if (isWithdrawalFromSubgraph(tx)) {
+    withdrawal = await mapWithdrawalToL2ToL1EventResult({
+      withdrawal: tx,
+      l1Provider: parentChainProvider,
+      l2Provider: chainProvider
+    })
+  } else {
+    if (isTokenWithdrawal(tx)) {
+      withdrawal = await mapTokenWithdrawalFromEventLogsToL2ToL1EventResult({
+        result: tx,
+        l1Provider: parentChainProvider,
+        l2Provider: chainProvider
+      })
+    } else {
+      withdrawal = await mapETHWithdrawalToL2ToL1EventResult({
+        event: tx,
+        l1Provider: parentChainProvider,
+        l2Provider: chainProvider
+      })
+    }
+  }
+
+  if (withdrawal) {
+    return transformWithdrawal(withdrawal)
+  }
+}
+
+function getProvider(chainId: ChainId) {
+  const rpcUrl =
+    rpcURLs[chainId] ?? getWagmiChain(chainId).rpcUrls.default.http[0]
+  return new StaticJsonRpcProvider(rpcUrl)
+}
+
+/**
+ * Fetches transaction history only, and only for a specific direction.
+ * The query could be e.g. deposits or withdrawals.
+ */
+const useTransactionListByDirection = (
+  direction: 'deposits' | 'withdrawals'
+) => {
+  const [transactions, setTransactions] = useState<DepositOrWithdrawal[][]>([])
+  const [page, setPage] = useState(0)
+  const [loading, setLoading] = useState(true)
+
+  const latestTransactions = useLatest(transactions)
```

Comment:
Curious to know - What benefit is using `useLatest` providing us here?

Reply:
I think I had a solution before that required this, but it should work without it now. I will remove and try.

Reply:
Seems fine without it, removed